crates/oxc_linter/src/config/mod.rs

```diff
@@ -0,0 +1,177 @@
+        .collect::<std::collections::HashSet<_>>();
+
+    let roles_hm = parse_rules(&file)
+        .unwrap()
+        .into_iter()
+        .map(|(plugin_name, rule_name, allow_warn_deny, config)| {
+            ((plugin_name, rule_name), (allow_warn_deny, config))
+        })
+        .collect::<std::collections::HashMap<_, _>>();
+
+    // `extends` provides the defaults
+    // `rules` provides the overrides
+    let m = RULES.clone().into_iter().filter_map(|rule| {
+        // Check if the extends set is empty or contains the plugin name
+        let in_extends = extends_hm.contains(rule.plugin_name());
+
+        // Check if there's a custom rule that explicitly handles this rule
+        let (is_explicitly_handled, is_explicitly_enabled, config) =
+            if let Some((policy, config)) = roles_hm.get(&(rule.plugin_name(), rule.name())) {
+                // Return true for handling, and also whether it's enabled or not
+                (true, policy.is_enabled(), config)
+            } else {
+                // Not explicitly handled
+                (false, false, &None)
+            };
+
+        // The rule is included if it's in the extends set and not explicitly disabled,
+        // or if it's explicitly enabled
+        if (in_extends && !is_explicitly_handled) || is_explicitly_enabled {
+            Some(rule.read_json(config.cloned()))
+        } else {
+            None
+        }
+    });
+
+    Ok(m.collect::<Vec<_>>())
+}
+
+fn parse_extends(root_json: &Value) -> Result<Option<Vec<&'static str>>, ()> {
+    let Some(extends) = root_json.get("extends") else {
+        return Ok(None);
+    };
+
+    let extends_obj = match extends {
+        Value::Array(v) => v,
+        _ => return Err(()),
+    };
+
+    let extends_rule_groups = extends_obj
+        .iter()
+        .filter_map(|v| {
+            let v = match v {
+                Value::String(s) => s,
+                _ => return None,
+            };
+
+            if let Some(m) = EXTENDS_MAP.get(v.as_str()) {
+                return Some(*m);
+            }
+
+            None
+        })
+        .collect::<Vec<_>>();
+
+    Ok(Some(extends_rule_groups))
+}
+
+#[allow(clippy::type_complexity)]
+fn parse_rules(
+    root_json: &Value,
+) -> Result<Vec<(&str, &str, AllowWarnDeny, Option<&Value>)>, Error> {
+    let Value::Object(rules_object) = root_json else { return Ok(vec![]) };
+
+    let Some(Value::Object(rules_object)) = rules_object.get("rules") else { return Ok(vec![]) };
+
+    rules_object
+        .iter()
+        .map(|(key, value)| {
+            let (plugin_name, name) = parse_rule_name(key);
+
+            let (rule_severity, rule_config) = resolve_rule_value(value)?;
+
+            Ok((plugin_name, name, rule_severity, rule_config))
+        })
+        .collect::<Result<Vec<_>, Error>>()
+}
+
+// maps the names oxc uses to the names eslint uses
+pub const RULE_NAMES_MAP: Map<&'static str, &'static str> = phf_map! {
+    "eqeqeq" => "eq-eq-eq"
```

Comment:
This is weird, how did it end up with `eqeqeq`? let's try and fix this on main and remove this logic from here. I want exact matches for maintainability.

Reply:
Oh I named it `EqEqEq` instead of `Eqeqeq` ... Let's fix this on main 😄